lib/matplotlib/tests/test_axes.py

```diff
@@ -260,6 +260,29 @@ def test_strmethodformatter_auto_formatter():
     assert ax.yaxis.get_minor_formatter().fmt == targ_strformatter.fmt
 
 
+def test_twinning():
+    # plot the axis
+    ax1 = plt.axes()
+
+    # get the currently position
+    pos1 = ax1.get_position()
+    # pos2 takes pos1 coords, and, alter the height by multiplying it by 0.6
+    pos2 = mtransforms.Bbox.from_extents([pos1.x0, pos1.y0,  pos1.width, pos1.height * 0.6])
+    
+    # twiny() ax1
+    ax1.twiny()
+
+    # set a new position
+    ax1.set_position(pos2)
+
+    # it gets the position, in a np.array, of ax1 after the twiny and the set_position
+    array1 = np.any(mtransforms.Bbox.get_points(ax1.get_position()))
+    # it gets the np.array of the point of pos2
+    array2 = np.any(mtransforms.Bbox.get_points(pos2))
+
+    # Assert that the positions are the same, and the twinning don't change them
+    assert array1 == array2
```

Comment:
array1 and array2 are both booleans after using np.any  (most likely `True`). You're asserting that True == True, which passes. Based on the variable names, I think you're expecting them to be arrays.

I think you're trying to assert that mtransforms.Bbox.get_points(pos2) is an array which is equal to mtransforms.Bbox.get_points(ax1.get_position()). These are both arrays and I have checked locally and they are equal to one another in this test.

When checking if NumPy arrays are equal to one another, you should use a NumPy function especially designed for this. You can use numpy.allclose (https://numpy.org/doc/stable/reference/generated/numpy.allclose.html) for values within a tolerance or numpy.array_equal (https://numpy.org/doc/stable/reference/generated/numpy.array_equal.html) to test that arrays are exactly equal.

Reply:
Sure, this is exactly what I wanted to do, because the twiny() were causing a dislocation in the axes that was not intended, and I setted manualy the position, twiny() it, and checked if the position matches to what it should. But I did not know how to compare this 2 arrays, now I think I know how can I do it.